File: leetcode/python/0455_assign_cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        
        i = 0
        for j in range(0, len(s)):
            if s[j] >= g[i]:
                i += 1

            if i == len(g): return i

        return i

File: leetcode/java/0455_assign_cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        if (s.length == 0) return 0;
        Arrays.sort(g);
        Arrays.sort(s);

        // loop through g and s using i and j respectively:
        // if (s[j] >= g[i]): res++, i++, j++
        // else if (s[j] < g[i]): j++
        // return res
        int res = 0;
        for (int i = 0, j = 0; i < g.length && j < s.length; ) {
            if (s[j] >= g[i]) {
                res++;
                i++;
                j++;
            } else if (s[j] < g[i]) j++;
        }

        return res;
    }
}